### PR TITLE
Enable cancel autoupload action for drafts and update labels

### LINKED ISSUE #10465: PostList: Upload media in offline and save draft
### Expected behavior
1. Open an existing draft
2. Turn on airplane mode
3. Add an image into the content
4. Click on "Save" under more in the toolbar

You should see 
- "Couldn't perform operation. We'll try again later" (yellow)
- "Cancel" (yellow)
- "Retry" (under more, grey)

### Actual behavior
We don't show the "Cancel" action on drafts. However, I believe we should show it as the post might not be a draft in the remote so the user might still want to cancel the auto-upload action. Wdyt @osullivanchris?

Note: https://github.com/wordpress-mobile/WordPress-Android/pull/10446 needs to be merged first.

### PR DESCRIPTION
**Pull request by malinajirka**
Fixes #10465

Review instructions:
1. Turn this Draft PR into a real PR -> We haven't agreed that this is the solution we want to go with, so I created just a draft PR. Feel free to turn it into a real PR if you like the solution.
2. Add a "copy review" label
3. Review the PR
4. Update target branch to `master-auto-upload` when https://github.com/wordpress-mobile/WordPress-Android/pull/10446 is merged
5. Approve and merge the PR

Note:
Initial goal of #10465 was to handle scenario when the user adds a media to a draft in offline - allow them to cancel the auto-upload action. However, I realized we probably should allow them to cancel the auto-upload even when there isn't any failed media upload (more info https://github.com/wordpress-mobile/WordPress-Android/issues/10465#issuecomment-534040153). 
I decided to show the cancel action on all drafts which are pending auto-upload (the changes were confirmed by the user by clicking on the "Save" button). I also decided to update the labels so the UX is consistent.

To test:
1. Open an existing draft
2. Turn on airplane mode
3. Add an image
4. Click on "Save"
5. Notice the item in the post list has yellow "Couldn't perform operation. We'll try again later" label and yellow "Cancel" action.
-----
1. Open an existing draft
2. Turn on airplane mode
3. Edit the content
4. Click on "Save"
5. Notice the item has yellow "Draft will be saved to the server next time your device is online" and "Cancel" action
6. Click on cancel
7. Notice snackbar with "Changes will not be saved to the server" is shown
(this might not be ideal as the changes will actually be saved using remote-auto-save -> not sure if we should worry about it or just keep it simple)

Update release notes:

- [ x ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Copy Review Request

Please see https://github.com/wordpress-mobile/WordPress-Android/pull/10512#issuecomment-534782110 for the summary.

### CODE REVIEW

**Comment by oguzkocer** [comment]
When I ran the first test and then cancel, I had an error that wouldn't go away. I suspected it was related to another PR we already merged, so I've merged the master branch into this PR. That seems to have fixed the issue.

**TL;DR:** Merged master branch into this PR.

**Comment by shiki** [comment]
Thanks, @oguzkocer. For some parts, we relied on Chris for the copy, and on some that we weren't so sure of, we asked for copy review. As for #10446, I definitely forgot to ask for a copy review. There is room for improvement in terms of consistency here. 😐 

I believe Jirka wanted a copy review for this since the new strings haven't been discussed before so I'm requesting it here... 

## Copy Review Request

Hi! 👋 We're adding these new strings in this task and we'd love your feedback. 🙂 

The message “Draft will be saved to the server next time your device is online“ is shown after creating a draft while offline and pressing More → Save. 



The message “Changes will not be saved to the server” is shown in the Snackbar after pressing the Cancel button shown above. Pressing this button informs the app to not automatically upload the post.

**Comment by oguzkocer** [comment]
We are merging this PR to be able to test the final behavior of the post offline experience _before_ we get a response for the copy review. However, we are only merging this to a feature branch meaning that it'll not be shipped to the users yet. We'll address any copy change request in a new PR.

**Comment by benhuberman** [comment]
Thanks @malinajirka @shiki -- the [comment](https://github.com/wordpress-mobile/WordPress-Android/pull/10512#issuecomment-534782110) with the specific copy review questions/instructions is always appreciated and really helps us focus on what the team actually needs in a giver PR. 

As for the language itself, there's some room to make it a bit warmer/friendlier and user-focused. 

Instead of "Draft will be saved to the server next time your device is online" I suggest:

> You seem to be offline. We'll save your draft as soon as your internet connection is back.

Instead of "Changes will not be saved to the server," I suggest:

> We won't save the latest changes to your draft.

Giving the other strings listed above a quick glance, I think there's an opportunity to tweak those as well; let me know if this should wait for a different PR, or if there's another way to tackle these soon.

**Comment by malinajirka** [comment]
Thank you @benhuberman !

I completely agree the language isn't as warm/friendly as it could. However, if I recall correctly we were trying to to come up with as concise messages as possible during the previous editorial review. We opted for more concise labels over friendlier labels. The labels on post list items were originally designed for 1-3 words and even `Post will be saved next time your device is online/...`  is "too" long. We weren't able to come up with anything shorter which would still make sense. 

I'm open to improving all the messages from the screenshot above. If we can come up with a friendlier but still concise copy, that'd be great. Either case I believe we should definitely keep them consistent. Wdyt? 

>We won't save the latest changes to your draft.

This copy sounds great! The SnackBar messages aren't as space limited and the proposed copy is still short enough. Do you think you could come up with a suggestion for all the  `Changes will not by *" messages so we can keep the consistent? Thanks!

**Comment by osullivanchris** [comment]
> You seem to be offline. We'll save your draft as soon as your internet connection is back.

While the copy in @benhuberman's suggestion quoted above seems quite nice for this scenario, I don't really like mixing "device is offline" with "internet connection is back". I think we should have one consistent way through all the messaging that we tell someone they're offline. There's lots of words that can mean the same thing between saying you/your device, online or connection, etc and we seem to have landed on "X will happen when next time your device is online" as a general pattern.

So although "Draft will be saved to the server next time your device is online" in the original screenshot looks a little awkward. I like the consistency of the  "...next time your device is online" part. What about something like this as a middle ground:

> We'll save your draft next time your device is online


**BIGGER QUESTION**
Regards the cancel action. I wonder what the user intent is here. Why would a user want to have something only locally on their device? I understand why someone would want to cancel publishing something when they come online. They want to control that situation and not leave it to be automated.

However, writing out a user scenario
* I'm not ready to publish my post. I explicitly hit 'Save' as I want to save the changes but not publish it yet
* My device is offline, so the save is only local on my device
* We tell the user "Draft will be saved to the server next time your device is online" and "Cancel" action
* First question - does the user understand the difference between saved locally and saved to server. Maybe we should be saying something like "You're offline. We've saved your post but only on this device for now" or something like that (I know this contradicts my earlier point but asking a broader question here)
* Lets assume that first question is not a problem. The user understands that the draft will be saved next time they come online
* What motivation would a user have to cancel this action? What would the user problem be? As I said I understand the problem with automatically publishing, where the user wants a lot of control. But automatically saving to the server seems very low risk. The only risk I could see is if the user does not want someone else with access to the same blog to see this post. But its a very specific scenario. And if the user does want to 'undo' would they not more likely want to get rid of the changes altogether rather than deliberately getting into a 'Local changes' state? It feels to me like 'Local changes' is an error state. And here we are almost trying to turn the error state into a desirable feature for the user :) 

Sorry to open up this discussion, I just don't know if there's a value having a cancel action for saving while offline, but I'm open to having my mind changed.

**Comment by malinajirka** [comment]
Thanks for jumping in @osullivanchris ! ;)

I really like the copy you proposed - `We'll save your draft next time your device is online`. If @benhuberman won't reply before the freeze, we'll go with that for now. We can improve it if necessary later.

Regarding the cancel action:). Consider the following scenario.
1. I create a draft and save it
2. The draft is uploaded to the server
3. I come back to the draft two days later, I edit it (eg fix a couple of grammar mistakes) and save it (I'm offline)
4. I realize I also want to add an image which is on my computer so I open the post on my computer. I add an image, fix the same grammar mistakes and I polish the post and click on Publish
5. I need to go back to my device and cancel the Save action as otherwise I'd override (un-publish) the published + polished version of the post with the image.

 I realize the cancel action won't be used very often, perhaps never, but I still believe it'd be nice to have it. Tbh even for the sake of consistency -> Whenever we use `* next time your device is online` we offer a cancel action.

Wdyt?

**Comment by benhuberman** [comment]
I like your suggested tweak, @osullivanchris. I'd change one thing here, which is using "back" instead of "next time" -- it's shorter and clearer. So:

> We'll save your draft when your device is back online.

Re: the other snippets, I can circle back to them next week, most likely.

**Comment by osullivanchris** [comment]
Sounds like we have the copy sorted.

@malinajirka on the 'why' for the cancel action. Sure that proves there is a valid case for it. It seems like an edge case but as we seem to have a solution ready for it I'm ok to use it. So lets go ahead with that. It does feel quite complicated. But moreso its complicated for us to figure it out again, if the user does have the intent you described I think it will be clear what to do.

**Comment by malinajirka** [comment]
I've created [a draft PR](https://github.com/wordpress-mobile/WordPress-Android/pull/10542) with the proposed changes. We can continue discussing there the changes for other post statuses.
